torch/_tensor_docs.py

```diff
@@ -1890,19 +1890,132 @@ def add_docstr_all(method, docstr):
                r"""
 index_put_(indices, values, accumulate=False) -> Tensor
 
-Puts values from the tensor :attr:`values` into the tensor :attr:`self` using
-the indices specified in :attr:`indices` (which is a tuple of Tensors). The
+Puts values from the tensor ``values`` into the tensor :attr:`self` using
+the indices specified in ``indices`` (which is a tuple of Tensors). The
 expression ``tensor.index_put_(indices, values)`` is equivalent to
 ``tensor[indices] = values``. Returns :attr:`self`.
 
-If :attr:`accumulate` is ``True``, the elements in :attr:`values` are added to
+If :attr:`accumulate` is ``True``, the elements in ``values`` are added to
 :attr:`self`. If accumulate is ``False``, the behavior is undefined if indices
 contain duplicate elements.
 
 Args:
     indices (tuple of LongTensor): tensors used to index into `self`.
     values (Tensor): tensor of same dtype as `self`.
     accumulate (bool): whether to accumulate into self
+
+Details:
+
+Each of the tensors in ``indices`` is a set of coordinates into :attr:`self`.
+The first tensor contains coordinates into the first dimension, the second
+contains coordinates into the second dimension, etc. The number of tensors in
+``indices`` is the number of dimensions that each individual coordinate has.
+The tensors must either all have the same shape or their shapes must be
+broadcastable into a common shape (see :ref:`broadcasting-semantics`). One set
+of corresponding elements within the broadcasted index tensors represents one
+index coordinate. For instance::
+
+    indices = [
+        tensor([0, 4, 3, 2]),
+        tensor([5, 6, 8, 9])
+    ]
+
+In this example, ``indices`` contains four 2-D coordinates, (0, 5), (4, 6),
+(3, 8), and (2, 9).
+
+Here is an example with broadcasting::
```

Comment:
I don't think example with broadcasting is needed here, it's standard broadcasting semantics.